src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Parbad. All rights reserved.
+// Licensed under the GNU GENERAL PUBLIC License, Version 3.0. See License.txt in the project root for license information.
+
+using Parbad.Gateway.PayIr.Internal;
+
+namespace Parbad.Gateway.PayIr
+{
+    public static class PayIrVerifyAdditionalDataExtensions
+    {
+        public static PayIrVerifyAdditionalData GetPayIrAdditionalData(this IPaymentVerifyResult result)
+        {
+            var additionalData = new PayIrVerifyAdditionalData();
+
+            if (result.AdditionalData.TryGetValue(nameof(PayIrVerifyResponseModel.CardNumber), out var cardNumber))
```

Comment:
No need to set and get the data 4 times! Only 1 time is fine.
return (PayIrVerifyAdditionalData) result.AdditionalData[AdditionalDataKey];

Reply:
I don't get it. I want to add 4 different additional data.
You mean all those fields will be serialized?

Reply:
Yes I serialized the additional data before.
But in new version the AdditionalData is a dictionary of <string, object>.
I explained it bellow.
No need to touch it for now.